blackbird_python/blackbird/program.py

```diff
@@ -314,25 +313,46 @@ def serialize(self):
 
         script = ["name {}".format(self.name), "version {}".format(self.version)]
 
-        if self.target["name"] is not None:
-            array_insert += 1
-            options = ""
+        # add target and type to the script
+        for name, data in [("target", self.target), ("type", self.programtype)]:
+            if data["name"] is not None:
+                array_insert += 1
+                options = ""
 
-            if self.target["options"]:
-                # if the target has options, compile them into
-                # the expected syntax
-                option_strings = [
-                    "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
-                    for k, v in self.target["options"].items()
-                ]
-                options = " ({})".format(", ".join(option_strings))
+                if data["options"]:
+                    # if the target has options, compile them into
+                    # the expected syntax
+                    option_strings = [
+                        "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
+                        for k, v in data["options"].items()
+                    ]
+                    options = " ({})".format(", ".join(option_strings))
 
-            # add target metadata
-            script.append("target {}{}".format(self.target["name"], options))
+                # add target metadata
+                script.append("{} {}{}".format(name, data["name"], options))
 
         # line break
         script.append("")
 
+        # add variables to the script
+        if self.programtype["name"] == "tdm":
+            from .listener import NUMPY_TYPES  # pylint:disable=import-outside-toplevel
+            inv_type_map = {np.dtype(v).kind: k for k, v in NUMPY_TYPES.items()}
```

Comment:
(Minor suggestions:) this seems to be something that could be generally used, how about having it in `listener.py` (e.g., after the def of `NUMPY_TYPES`)?